pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -74,3 +74,219 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
```

Comment:
Question: How do we know which fields were optional in the flag based config? Are we relying on flag description?